chains/evm/cli/erc20/balance.go

```diff
@@ -57,21 +63,41 @@ func balance(cmd *cobra.Command, args []string, txFabric calls.TxFabric) error {
 	}
 
 	// erc20Addr, accountAddr
-	input, err := calls.PrepareERC20BalanceInput(erc20Addr, accountAddr)
+	input, err := calls.PrepareERC20BalanceInput(accountAddr)
 	if err != nil {
 		log.Error().Err(fmt.Errorf("prepare input error: %v", err))
 		return err
 	}
 
-	txHash, err := calls.Transact(ethClient, txFabric, &erc20Addr, input, gasLimit)
+	msg := ethereum.CallMsg{
+		From: common.Address{},
+		To:   &erc20Addr,
+		Data: input,
+	}
+
+	out, err := ethClient.CallContract(context.TODO(), calls.ToCallArg(msg), nil)
+	if err != nil {
+		log.Error().Err(fmt.Errorf("call contract error: %v", err))
+		return err
+	}
+
+	if len(out) == 0 {
+		// Make sure we have a contract to operate on, and bail out otherwise.
+		if code, err := ethClient.CodeAt(context.Background(), erc20Addr, nil); err != nil {
+			return err
+		} else if len(code) == 0 {
+			return fmt.Errorf("no code at provided address %s", erc20Addr.String())
+		}
+	}
+
+	res, err := calls.PrepareERC20BalanceOutput(out)
 	if err != nil {
-		log.Error().Err(fmt.Errorf("send input error: %v", err))
+		log.Error().Err(fmt.Errorf("prepare output error: %v", err))
 		return err
 	}
-	log.Debug().Msgf("tx hash: %v", txHash.Hex())
 
-	// determine balance
+	balance := *abi.ConvertType(res[0], new(big.Int)).(*big.Int)
```

Comment:
put it inside PrepareERC20BalanceOutput function, and name it like ParseERC20BalanceOutput

Reply:
done